pkg/services/librarypanels/librarypanels_test.go

```diff
@@ -178,12 +178,240 @@ func TestGetAllLibraryPanels(t *testing.T) {
+	testScenario(t, "When an admin tries to update a library panel that does not exist, it should fail",
+		func(t *testing.T, sc scenarioContext) {
+			cmd := updateLibraryPanelCommand{}
+			response := sc.service.updateHandler(sc.reqContext, cmd)
+			require.Equal(t, 404, response.Status())
+		})
+
+	testScenario(t, "When an admin tries to update a library panel that exists, it should succeed",
+		func(t *testing.T, sc scenarioContext) {
+			command := getCreateCommand(1, "Text - Library Panel")
+			response := sc.service.createHandler(sc.reqContext, command)
+			require.Equal(t, 200, response.Status())
+
+			var result libraryPanelResult
+			err := json.Unmarshal(response.Body(), &result)
+			require.NoError(t, err)
+
+			cmd := updateLibraryPanelCommand{
+				FolderID: 2,
+				Name:     "Panel - New name",
+				Model: []byte(`
+								{
+								  "datasource": "${DS_GDEV-TESTDATA}",
+								  "id": 1,
+								  "name": "Model - New name",
+								  "type": "text"
+								}
+							`),
+			}
+			sc.reqContext.ReplaceAllParams(map[string]string{":uid": result.Result.UID})
+			response = sc.service.updateHandler(sc.reqContext, cmd)
+			require.Equal(t, 200, response.Status())
+			err = json.Unmarshal(response.Body(), &result)
+			require.NoError(t, err)
+			require.Equal(t, int64(2), result.Result.FolderID)
+			require.Equal(t, "Panel - New name", result.Result.Name)
+			require.Equal(t, "Model - New name", result.Result.Model["name"])
+		})
+
+	testScenario(t, "When an admin tries to update a library with folder only, it should change folder successfully and return correct result",
+		func(t *testing.T, sc scenarioContext) {
+			command := getCreateCommand(1, "Text - Library Panel")
+			response := sc.service.createHandler(sc.reqContext, command)
+			require.Equal(t, 200, response.Status())
+
+			var existing libraryPanelResult
+			err := json.Unmarshal(response.Body(), &existing)
+			require.NoError(t, err)
+
+			cmd := updateLibraryPanelCommand{
+				FolderID: 100,
+			}
+			sc.reqContext.ReplaceAllParams(map[string]string{":uid": existing.Result.UID})
+			response = sc.service.updateHandler(sc.reqContext, cmd)
+			require.Equal(t, 200, response.Status())
+			var result libraryPanelResult
+			err = json.Unmarshal(response.Body(), &result)
+			require.NoError(t, err)
+			require.Equal(t, existing.Result.ID, result.Result.ID)
+			require.Equal(t, existing.Result.OrgID, result.Result.OrgID)
+			require.Equal(t, int64(100), result.Result.FolderID)
+			require.Equal(t, existing.Result.Name, result.Result.Name)
+			require.Equal(t, existing.Result.Model["name"], result.Result.Model["name"])
+			require.Equal(t, existing.Result.Created.UTC().Format(time.RFC3339), result.Result.Created.UTC().Format(time.RFC3339))
+			require.Equal(t, existing.Result.CreatedBy, result.Result.CreatedBy)
+			require.True(t, existing.Result.Updated.UTC().Before(result.Result.Updated.UTC()))
+			require.Equal(t, existing.Result.UpdatedBy, result.Result.UpdatedBy)
```

Comment:
Safer and more succinct to use go-cmp here.

Reply:
Fixed